lib/gems/pending/util/object_storage/miq_ftp_storage.rb

```diff
@@ -0,0 +1,92 @@
+require 'util/miq_ftp_lib'
+require 'util/miq_object_storage'
+require 'logger'
+
+class MiqFtpStorage < MiqObjectStorage
+  include MiqFtpLib
+
+  FTP_CHUNKSIZE = ::Net::FTP::DEFAULT_BLOCKSIZE
+
+  attr_reader :uri, :username, :password
+
+  class << self
```

Comment:
Going to remove this whole block as we have `MiqObjectStorage#logger`, so I will just make use of that.